test/test_decoding.py

```diff
@@ -33,6 +34,24 @@ class SomeClass(Serializable):
     assert SomeClass.loads(b.dumps()) == b
 
 
+def test_literal_decoding():
+    @dataclass
+    class SomeClass(Serializable):
+        x: Literal["a", "b", "c"] = "a"
+
+    # This test should fail if there's a warning on decoding- previous versions
+    # have raised a UserWarning when decoding a literal, of the form:
+    # Unable to find a decoding function for annotation typing.Literal['a', 'b', 'c']
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
I guess the test to replicate the issue would have used `pytest.warns`, right?
Since we expect the test not to warn anymore, I think it's fine to make this a comment.

Reply:
Ok! Feel free to do that yourself, or I can do it in a bit